components/dashboard/src/settings/Notifications.tsx

```diff
@@ -1,9 +1,62 @@
+import { useContext } from "react";
 import { SettingsPage } from "./SettingsPage";
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
 
 export default function Notifications() {
+    const ctx = useContext(UserContext);
+    const user = ctx.user;
+    const isTransactionalMail = !user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails;
+    const toggleTransactionalMail = async ()=>{
+        if (user) {
+            user.additionalData = {
```

Comment:
Maybe get `user.additionalData` inside the handler?

Getting it outside of it means that you will overwrite any `additionalData` changes that happened between:
- This component was rendered
- A checkbox was ticked

Proof of concept: https://youtu.be/bYD8s3yivEw (my IDE change gets overwritten)

Reply:
The user is cached within the app. So in order to solve the (unlikely) case that someone changes settings in two browser tabs, we'd need to fetch the user from the server before altering it.